pkg/controller/hibernation/hibernation_controller_test.go

```diff
@@ -289,6 +293,158 @@ func TestReconcile(t *testing.T) {
+				testcd.WithCondition(hibernatingCondition(corev1.ConditionFalse, hivev1.RunningHibernationReason, 6*time.Hour)),
+			).Options(
+				testcd.InstalledTimestamp(time.Now().Add(-10 * time.Hour))).Build(),
+			expectRequeueAfter: true,
+			expectedPowerState: "",
+		},
+		{
+			name: "cluster stuck resuming condition due for hibernate",
+			cd: cdBuilder.GenericOptions(
+				testgeneric.WithAnnotation(constants.HibernateAfterAnnotation, "8h")).Options(
+				// TODO: check false below
+				testcd.WithCondition(hibernatingCondition(corev1.ConditionFalse, hivev1.ResumingHibernationReason, 9*time.Hour)),
+			).Options(
+				testcd.InstalledTimestamp(time.Now().Add(-10 * time.Hour))).Build(),
+			expectedPowerState: hivev1.HibernatingClusterPowerState,
+		},
+		{
+			name: "cluster failed resuming condition due for hibernate",
+			cd: cdBuilder.GenericOptions(
+				testgeneric.WithAnnotation(constants.HibernateAfterAnnotation, "8h")).Options(
+				// TODO:
+				testcd.WithCondition(hibernatingCondition(corev1.ConditionFalse, hivev1.FailedToStartHibernationReason, 9*time.Hour)),
+			).Options(
+				testcd.InstalledTimestamp(time.Now().Add(-10 * time.Hour))).Build(),
+			expectedPowerState: hivev1.HibernatingClusterPowerState,
+		},
+		{
+			name: "cluster waking from hibernate",
+			setupActuator: func(actuator *mock.MockHibernationActuator) {
+				actuator.EXPECT().MachinesRunning(gomock.Any(), gomock.Any(), gomock.Any()).Times(1).Return(false, nil)
+			},
+			cd: cdBuilder.GenericOptions(
+				testgeneric.WithAnnotation(constants.HibernateAfterAnnotation, "4h")).Options(
+				testcd.WithCondition(hibernatingCondition(corev1.ConditionTrue, hivev1.ResumingHibernationReason, 8*time.Hour)),
+			).Options(
+				testcd.InstalledTimestamp(time.Now().Add(-10*time.Hour)),
+				o.shouldRun).Build(),
+			expectedPowerState: hivev1.RunningClusterPowerState,
+			expectRequeueAfter: true,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			mockActuator := mock.NewMockHibernationActuator(ctrl)
+			mockActuator.EXPECT().CanHandle(gomock.Any()).AnyTimes().Return(true)
+			if test.setupActuator != nil {
+				test.setupActuator(mockActuator)
+			}
+			mockBuilder := remoteclientmock.NewMockBuilder(ctrl)
+			mockCSRHelper := mock.NewMockcsrHelper(ctrl)
+			actuators = []HibernationActuator{mockActuator}
+			c := fake.NewFakeClientWithScheme(scheme, test.cd)
+
+			reconciler := hibernationReconciler{
+				Client: c,
+				logger: log.WithField("controller", "hibernation"),
+				remoteClientBuilder: func(cd *hivev1.ClusterDeployment) remoteclient.Builder {
+					return mockBuilder
+				},
+				csrUtil: mockCSRHelper,
+			}
+			result, err := reconciler.Reconcile(reconcile.Request{
+				NamespacedName: types.NamespacedName{Namespace: namespace, Name: cdName},
+			})
+			if test.expectRequeueAfter {
+				assert.True(t, result.RequeueAfter > 0)
```

Comment:
Please add messages to asserts and requires.
```suggestion
				assert.True(t, result.RequeueAfter > 0, "expected positive requeue after")
```